src/vs/workbench/contrib/url/common/url.contribution.ts

```diff
@@ -34,5 +36,107 @@ export class OpenUrlAction extends Action {
-	.registerWorkbenchAction(new SyncActionDescriptor(OpenUrlAction, OpenUrlAction.ID, OpenUrlAction.LABEL), 'Open URL', localize('developer', "Developer"));
\ No newline at end of file
+Registry.as<IWorkbenchActionRegistry>(ActionExtensions.WorkbenchActions).registerWorkbenchAction(
+	new SyncActionDescriptor(OpenUrlAction, OpenUrlAction.ID, OpenUrlAction.LABEL),
+	'Open URL',
+	localize('developer', 'Developer')
+);
+
+const configureTrustedDomainsHandler = (
+	quickInputService: IQuickInputService,
+	storageService: IStorageService,
+	domainToConfigure?: string
+) => {
+	let trustedDomains: string[] = [];
+	try {
+		trustedDomains = JSON.parse(storageService.get('http.trustedDomains', StorageScope.GLOBAL, '[]'));
+	} catch (err) { }
+
+	const domainQuickPickItems: IQuickPickItem[] = trustedDomains
+		.filter(d => d !== '*')
+		.map(d => {
+			return {
+				type: 'item',
+				label: d,
+				picked: true,
+			};
+		});
+
+	const specialQuickPickItems: IQuickPickItem[] = [
+		{
+			type: 'item',
+			label: localize('allowAllLinks', 'Allow all links to be open without protection'),
+			picked: trustedDomains.indexOf('*') !== -1
+		}
+	];
+
+	let domainToConfigureItem: IQuickPickItem | undefined = undefined;
+	if (domainToConfigure) {
+		domainToConfigureItem = {
+			type: 'item',
+			label: domainToConfigure,
+			picked: true,
+			description: localize('trustDomainAndOpenLink', 'Trust domain and open link')
+		};
+		specialQuickPickItems.push(<IQuickPickItem>domainToConfigureItem);
+	}
+
+	const quickPickItems: (IQuickPickItem | IQuickPickSeparator)[] = domainQuickPickItems.length === 0
+		? specialQuickPickItems
+		: [...specialQuickPickItems, { type: 'separator' }, ...domainQuickPickItems];
+
+	return quickInputService.pick(quickPickItems, {
+		canPickMany: true,
+		activeItem: domainToConfigureItem
+	}).then(result => {
+		if (result) {
+			const pickedDomains = result.map(r => r.label);
+			storageService.store('http.trustedDomains', JSON.stringify(pickedDomains), StorageScope.GLOBAL);
+
+			return pickedDomains;
+		}
+
+		return [];
+	});
+};
+
+export class ConfigureTrustedDomainsAction extends Action {
+
+	static readonly ID = 'workbench.action.configureTrustedDomains';
+	static readonly LABEL = localize('configureTrustedDomains', "Configure Trusted Domains");
+
+	constructor(
+		id: string,
+		label: string,
+		@IQuickInputService private readonly quickInputService: IQuickInputService,
+		@IStorageService private readonly storageService: IStorageService
+	) {
+		super(id, label);
+	}
+
+	run(): Promise<any> {
+		return configureTrustedDomainsHandler(this.quickInputService, this.storageService);
+	}
+}
+
+Registry.as<IWorkbenchActionRegistry>(ActionExtensions.WorkbenchActions).registerWorkbenchAction(
+	new SyncActionDescriptor(
+		ConfigureTrustedDomainsAction,
+		ConfigureTrustedDomainsAction.ID,
+		ConfigureTrustedDomainsAction.LABEL
+	),
+	'Configure Trusted Domains'
+);
+CommandsRegistry.registerCommand({
```

Comment:
@jrieken I'm not quite sure if this is the right code. `Registry.as<IWorkbenchActionRegistry>(ActionExtensions.WorkbenchActions).registerWorkbenchAction` doesn't make the command available in command registry to be run with args, so I had to register it separately.

Reply:
you don't need any of the action stuff. it is enough to register the command (`registerCommand`) and make it show in the command palette it's enough to call `MenuRegistry.appendMenuItem` with the command palette menu id, like so: https://github.com/microsoft/vscode/blob/c8d44e24dd250d416475a10efe64e35c9e2c65b3/src/vs/editor/contrib/gotoError/gotoError.ts#L509

Reply:
Updated. Thanks for the pointer.